cfgrammar/src/lib/span.rs

```diff
@@ -126,12 +126,21 @@ impl NewlineToLineColCache {
     ) -> Option<(usize, usize)> {
         let line_num = self.byte_to_line_num(byte);
         line_num.map(|line_num| {
-            if src_pos + byte == src.len() {
-                let line_off = self.newlines.last().unwrap();
-                return (self.newlines.len(), src[*line_off..].chars().count() + 1);
+            let line_byte = self.line_num_to_byte(line_num);
```

Comment:
It strikes me that this function isn't taking `trailing_bytes` into account, and I feel like probably needs to, i'll have to think it over

Reply:
I think being driven by tests would be a good thing here!